src/analysis/vector/qgszonalstatistics.cpp

```diff
@@ -31,12 +31,37 @@
 #include <QFile>
 
 QgsZonalStatistics::QgsZonalStatistics( QgsVectorLayer *polygonLayer, QgsRasterLayer *rasterLayer, const QString &attributePrefix, int rasterBand, QgsZonalStatistics::Statistics stats )
-  : mRasterLayer( rasterLayer )
+  : QgsZonalStatistics( polygonLayer,
+                        rasterLayer ? rasterLayer->dataProvider() : nullptr,
+                        rasterLayer ? rasterLayer->crs() : QgsCoordinateReferenceSystem(),
+                        rasterLayer ? rasterLayer->rasterUnitsPerPixelX() : 0,
+                        rasterLayer ? rasterLayer->rasterUnitsPerPixelY() : 0,
+                        attributePrefix,
+                        rasterBand,
+                        stats )
+{
+}
+
+QgsZonalStatistics::QgsZonalStatistics( QgsVectorLayer *polygonLayer, QgsRasterInterface *rasterInterface,
+                                        const QgsCoordinateReferenceSystem &rasterCrs, double rasterUnitsPerPixelX, double rasterUnitsPerPixelY, const QString &attributePrefix, int rasterBand, QgsZonalStatistics::Statistics stats )
+  : mRasterInterface( rasterInterface )
+  , mRasterCrs( rasterCrs )
+  , mCellSizeX( rasterUnitsPerPixelX )
+  , mCellSizeY( rasterUnitsPerPixelY )
   , mRasterBand( rasterBand )
   , mPolygonLayer( polygonLayer )
   , mAttributePrefix( attributePrefix )
   , mStatistics( stats )
-{}
+{
+  if ( mCellSizeX < 0 )
+  {
+    mCellSizeX = -mCellSizeX;
+  }
```

Comment:
Would it be possible to do something like `, mCellSizeX( std::abs( rasterUnitsPerPixelX ) )` instead?

(same question about `mCellSizeY`)

Reply:
Fixed